projects/ui/src/state/beanstalk/field/morning.ts

```diff
@@ -0,0 +1,145 @@
+  const sunriseBlock = season.sunriseBlock;
+  const deltaBlocks = morningBlock.minus(sunriseBlock);
+
+  /// Temperature Season Data
+  const seasonData = temperatureQuery.data?.seasons;
+  const currentSeason = season.current;
+  const first = seasonData?.[0];
+
+  /// -------------------------------------
+  /// Callbacks
+
+  const fetchSoil = useCallback(async () => {
+    if (!beanstalk) {
+      console.debug(`[beanstalk/field/morning] fetch: contract undefined`);
+      return;
+    }
+    try {
+      const _soil = await beanstalk.totalSoil().then(tokenResult(BEAN));
+      console.debug('[beanstalk/field/morning] fetch: soil', _soil.toNumber());
+      if (!soil.eq(_soil)) {
+        dispatch(updateTotalSoil(_soil));
+      }
+    } catch (err) {
+      console.debug('[beanstalk/field/morning] fetch FAILED', err);
+    }
+  }, [soil, beanstalk, dispatch]);
+
+  /// -------------------------------------
+  /// Effects
+
+  // If it is morning, then we fetch the soil every 4 seconds
+  useEffect(() => {
+    if (!morning.isMorning) return;
+
+    const soilUpdateInterval = setInterval(() => {
+      const now = getNowRounded();
+
+      const remaining = getDiffNow(next, now).as('seconds');
+      if (remaining % 4 === 0) {
+        fetchSoil();
+      }
+    }, 1000);
+
+    return () => {
+      clearInterval(soilUpdateInterval);
+    };
+  }, [fetchSoil, morning.isMorning, next]);
+
+  /**
+   * Refetch the field every 2 seconds for updates if:
+   *
+   * It is not morning:
+   * - If the max temperature is not equal to the scaled temperature
+   *
+   * OR
+   *
+   * If it is the morning:
+   *    We are in the 1st interval of the morning & the scaled temperature is not equal to 1.
+   *    The temperature of the 1st interval of the morning is always 1%.
+   */
+  const shouldUpdateField = (() => {
+    if (!morning.isMorning) {
+      return !temperature.max.eq(temperature.scaled);
+    }
+    if (morning.isMorning) {
+      return deltaBlocks.isZero() && !temperature.scaled.eq(1);
+    }
+    return false;
+  })();
+
+  useEffect(() => {
+    if (!shouldUpdateField) return;
+
+    const interval = setInterval(() => {
+      console.debug('[beanstalk/field/morning]: Refetching field');
+      fetchBeanstalkField();
+    }, 2000);
+    return () => {
+      clearInterval(interval);
+    };
+  }, [fetchBeanstalkField, shouldUpdateField]);
+
+  /// If the user is behind 1 season, update the temperature data
+  /// This is to prevent the morning temperature chart & max temperature
+  /// chart from being out of sync.
+  useEffect(() => {
+    /// data not loaded yet
+    if (!first || currentSeason.lte(0)) return;
+
+    /// If we are behind 1 season, refetch
+    if (currentSeason.minus(first.season).eq(1)) {
+      console.debug('refetching temperature data...');
+      temperatureQuery.refetch({
+        season_lte: currentSeason.toNumber(),
+      });
+    }
+  }, [currentSeason, first, temperatureQuery]);
```

Comment:
I'd appreciate some help understanding why we're using subgraph data here. My understanding:

`currentSeason (season.current)` = from chain
`first` = the most recently updated season in the subgraph

Later we check if `(currentSeason - first.season) = 1`, i.e. that the current season has been updated on chain but we haven't polled for new season data from the subgraph.

Followups:
- What happens if things get out of sync by more than one season? Is this possible?
- Say that `currentSeason` changes right after the gm() call, but the subgraph hasn't yet created a new season entity. Would this cause the following error? 

currentSeason updated -> season delta is now 1 -> call temperatureQuery.refetch() -> query returns the same data since subgraph hasn't yet indexed the new season -> [now what?]

If `temperatureQuery` or `first` are modified, the effect will re-run and cause a loop until the subgraph catches up.
If they aren't modified, then it seems that the state would get stuck with a difference of 1 between temperatureQuery first and currentSeason

Reply:
Yeah, this is a good point. I updated it so that we fetch the maxTemperature only when transitioning out of the morning state. This will now be called only once, to prevent these above mentioned issues.